src/arkode/arkode_arkstep_nls.c

```diff
@@ -193,25 +190,29 @@ int arkStep_NlsInit(ARKodeMem ark_mem)
   step_mem->nls_fails = 0;
 
   /* set the linear solver setup wrapper function */
-  if (step_mem->lsetup)
+  if (step_mem->lsetup) {
```

Comment:
Could this entire code block be replaced by the simpler
```
if (step_mem->lsetup) {
  SUNCheckCall(SUNNonlinSolSetLSetupFn(step_mem->NLS, arkStep_NlsLSetup));
} else {
  SUNCheckCall(SUNNonlinSolSetLSetupFn(step_mem->NLS, NULL));
}
```

Reply:
I cant see the whole block in the dif, but Ill check.

Reply:
Likely not, due to the same reasoning as above wrt `arkProcessError`